homeassistant/components/media_player/denonavr.py

```diff
@@ -147,6 +160,24 @@ def __init__(self, receiver):
         self._frequency = self._receiver.frequency
         self._station = self._receiver.station
 
+        self._sound_mode_support = sound_mode_support
+        if sound_mode_support:
+            self._sound_mode = self._receiver.sound_mode
+            self._sound_mode_raw = self._receiver.sound_mode_raw
+            if sound_mode_dict is None:
```

Comment:
If we can fetch it from the device, we should not allow it to be configured. Integrations should only represent the raw information that we receive from the device.

Reply:
@balloob the sound_mode_dict can not be fetched from the device (receiver) itself, the denonavr library contains an default sound_mode_dict that is hard coded in the denonavr library. However that dictionary is based upon the data of about 10 diffrent models of denon receivers (from which the information xml files are available in the denonavr git repositry, but their could potentially be other values available for newer or very old receivers.
This allows for flexibility for these older or newer receivers.

Reply:
I think that we should not include this config option because it will not motivate people to update the dictionary upstream for old and new receivers, resulting in all users having to collect the same information.